include/CLI/TypeTools.hpp

```diff
@@ -961,20 +961,18 @@ bool integral_conversion(const std::string &input, T &output) noexcept {
         output = (output_sll < 0) ? static_cast<T>(0) : static_cast<T>(output_sll);
         return (static_cast<std::int64_t>(output) == output_sll);
     }
-    // remove locale-specific group separators
-    char group_separator = std::use_facet<std::numpunct<char>>(std::locale()).thousands_sep();
-    if(input.find_first_of(group_separator) != std::string::npos) {
-        std::string nstring = input;
-        nstring.erase(std::remove(nstring.begin(), nstring.end(), group_separator), nstring.end());
-        return integral_conversion(nstring, output);
-    }
-    // remove separators
-    if(input.find_first_of("_'") != std::string::npos) {
-        std::string nstring = input;
-        nstring.erase(std::remove(nstring.begin(), nstring.end(), '_'), nstring.end());
-        nstring.erase(std::remove(nstring.begin(), nstring.end(), '\''), nstring.end());
-        return integral_conversion(nstring, output);
+    // remove separators if present
+    auto group_separators = get_group_separators();
+    if(input.find_first_of(group_separators) != std::string::npos) {
+        for(auto &separator : group_separators) {
+            if(input.find_first_of(separator) != std::string::npos) {
+                std::string nstring = input;
+                nstring.erase(std::remove(nstring.begin(), nstring.end(), separator), nstring.end());
+                return integral_conversion(nstring, output);
+            }
+        }
```

Comment:
to reduce the recursion depth:
```suggestion
        std::string nstring = input;
        for(auto &separator : group_separators) {
            if(input.find_first_of(separator) != std::string::npos) {
                nstring.erase(std::remove(nstring.begin(), nstring.end(), separator), nstring.end());
            }
        }
        return integral_conversion(nstring, output);
```

Reply:
Thanks

Reply:
there are two more places with similar code